### PR TITLE
Convert OkHttpClient to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/OkHttpClient.kt [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+  private val writeTimeout: Int = builder.writeTimeout
+  private val pingInterval: Int = builder.pingInterval
+  private val internalCache: InternalCache? = builder.internalCache
+  private val certificateChainCleaner: CertificateChainCleaner?
+
+  constructor() : this(Builder())
+
+  init {
+    var isTLS = false
+    for (spec in connectionSpecs) {
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/OkHttpClient.kt]: Is this more idiomatic?

```
isTLS = connectionSpecs.any { it.isTLS }
```

File: okhttp/src/main/java/okhttp3/OkHttpClient.kt [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+    }
+
+    if (sslSocketFactory != null) {
+      Platform.get().configureSslSocketFactory(sslSocketFactory)
+    }
+
+    this.certificatePinner = builder.certificatePinner
+        .withCertificateChainCleaner(certificateChainCleaner)
+
+    check(null !in (interceptors as List<Interceptor?>)) {
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/OkHttpClient.kt]: Do we really need this check?  Should we trust the non null types.

File: okhttp/src/main/java/okhttp3/OkHttpClient.kt [hunk 3]
```diff
@@ -0,0 +1,827 @@
...
+     * be used.
+     *
+     * @deprecated [SSLSocketFactory] does not expose its [X509TrustManager], which is a field that
+     *     OkHttp needs to build a clean certificate chain. This method instead must use reflection
+     *     to extract the trust manager. Applications should prefer to call
+     *     `sslSocketFactory(SSLSocketFactory, X509TrustManager)`, which avoids such reflection.
+     */
+    @Deprecated(
+        message = "Use the sslSocketFactory overload that accepts a X509TrustManager.",
+        level = ERROR
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/OkHttpClient.kt]: Nice!

File: okhttp/src/main/java/okhttp3/OkHttpClient.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/OkHttpClient.kt]: Unfortunately Java doesn't honor them. We've had bugs where users get nulls into these collections and the crashes are difficult to diagnose!

File: okhttp/src/main/java/okhttp3/OkHttpClient.kt (hunk 1 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/OkHttpClient.kt]: Yes! I wanna save for follow up. Would be excellent for IntelliJ to quick fix this for me.

**Pull request merged**